runtime/vm/JFRChunkWriter.hpp

```diff
@@ -193,6 +194,38 @@ class VM_JFRChunkWriter {
 		return false;
 	}
 
+	U_8 *reserveCheckpointSize() {
+		if (J9JAVAVM_COMPRESS_OBJECT_REFERENCES(_vm)) {
+			return _bufferWriter->getAndIncCursor(sizeof(U_32));
+		} else {
+			return _bufferWriter->getAndIncCursor(sizeof(U_64) + sizeof(U_8));
+		}
+	}
+
+	void writeCheckpointSize(U_8 *checkpointEventStart) {
+		if (J9JAVAVM_COMPRESS_OBJECT_REFERENCES(_vm)) {
+			_bufferWriter->writeLEB128PaddedU32(checkpointEventStart, _bufferWriter->getCursor() - checkpointEventStart);
+		} else {
+			_bufferWriter->writeLEB128PaddedU72(checkpointEventStart, _bufferWriter->getCursor() - checkpointEventStart);
+		}
+	}
+
+	U_8 *reserveEventSize() {
+		return _bufferWriter->getAndIncCursor(sizeof(U_64) + sizeof(U_8));
+	}
+
+	void writeEventSize(U_8 *eventStart) {
+		_bufferWriter->writeLEB128PaddedU72(eventStart, (U_64)(_bufferWriter->getCursor() - eventStart));
+	}
+
+	static U_8 *reserveEventSize(VM_BufferWriter *bufferWriter) {
+		return bufferWriter->getAndIncCursor(sizeof(U_64) + sizeof(U_8));
+	}
+
+	static void writeEventSize(VM_BufferWriter *bufferWriter, U_8 *eventStart) {
+		bufferWriter->writeLEB128PaddedU72(eventStart, (U_64)(bufferWriter->getCursor() - eventStart));
+	}
```

Comment:
Suggest neither of these functions is needed.

Reply:
Would you mind explaining why?

Reply:
Everywhere this function might be called, it could be replaced by `bufferWriter->writeEventSize(eventStart)`.

Reply:
A similar comment applies to `reserveEventSize()`.

Reply:
Id like to keep JFR-isms out of `bufferWriter` so we can use in the future for other purposes.

Reply:
I agree: My suggestion was in error. Instead, I suggest the non-static methods be implemented via the static methods:
```c
	U_8 *reserveEventSize()
	{
		return reserveEventSize(_bufferWriter);
	}

	void writeEventSize(U_8 *eventStart)
	{
		writeEventSize(_bufferWriter, eventStart);
	}
```